lib/rules/no-useless-return.js

```diff
@@ -0,0 +1,295 @@
+
+        return {
+
+            // Makes and pushs a new scope information.
+            onCodePathStart(codePath) {
+                scopeInfo = {
+                    upper: scopeInfo,
+                    uselessReturns: [],
+                    codePath,
+                };
+            },
+
+            // Reports useless return statements if exist.
+            onCodePathEnd() {
+                for (const node of scopeInfo.uselessReturns) {
+                    context.report({
+                        node,
+                        loc: node.loc,
+                        message: "Unnecessary return statement.",
+                        fix(fixer) {
+                            return isRemoveable(node) ? fixer.remove(node) : null;
+                        },
+                    });
+                }
+
+                scopeInfo = scopeInfo.upper;
+            },
+
+            // Initializes segments.
+            // NOTE: This event is notified for only reachable segments.
+            onCodePathSegmentStart(segment) {
+                const info = {
+                    uselessReturns: getUselessReturns([], segment.allPrevSegments),
+                    returned: false,
+                };
+
+                // Stores the info.
+                segmentInfoMap.set(segment, info);
+            },
+
+            // Adds ReturnStatement node to check whether it's useless or not.
+            ReturnStatement(node) {
+                if (node.argument || isInLoop(node) || isInFinally(node)) {
+                    return;
+                }
+
+                for (const segment of scopeInfo.codePath.currentSegments) {
+                    const info = segmentInfoMap.get(segment);
+
+                    info.uselessReturns.push(node);
+                    info.returned = true;
+                }
+                scopeInfo.uselessReturns.push(node);
+            },
+
+            // Registers for all statement nodes except FunctionDeclaration, BlockStatement, BreakStatement.
+            // Removes return statements of the current segments from the useless return statement list.
+            ClassDeclaration: markReturnStatementsOnCurrentSegmentsAsUsed,
+            ContinueStatement: markReturnStatementsOnCurrentSegmentsAsUsed,
+            DebuggerStatement: markReturnStatementsOnCurrentSegmentsAsUsed,
+            DoWhileStatement: markReturnStatementsOnCurrentSegmentsAsUsed,
+            EmptyStatement: markReturnStatementsOnCurrentSegmentsAsUsed,
+            ExpressionStatement: markReturnStatementsOnCurrentSegmentsAsUsed,
+            ForInStatement: markReturnStatementsOnCurrentSegmentsAsUsed,
+            ForOfStatement: markReturnStatementsOnCurrentSegmentsAsUsed,
+            ForStatement: markReturnStatementsOnCurrentSegmentsAsUsed,
+            IfStatement: markReturnStatementsOnCurrentSegmentsAsUsed,
+            ImportDeclaration: markReturnStatementsOnCurrentSegmentsAsUsed,
```

Comment:
I also don't know such environments :smiley: .
In this case, I guessed `ImportDeclaration`s behaves as similar to `VariableDeclaration`s. On the other hand, even if `globalReturn: true` does not support `import`, this will not trigger false positive/negative.